### DISCUSSION TITLE
how to obtain whole MS2 by DIA

### DISCUSSION THREAD

**Discussion by CassyGu** [vdemichev/DiaNN#587] [Q&A]
Hi, Vadim,
The file of report.XIC.tsv only showed the matched b,y ions from individual precursors，while whole fragments were not shown;
Is there any method or DIANN parameters to show all fragment ions??

Thanks
Cassy

**Comment by vdemichev** [comment]
Hi Cassy,

All fragments considered are showed when using --vis, not just matched ones.

Best,
Vadim

**Reply by vdemichev** [reply]
That is all fragments in the library. To consider all theoretical fragments, need to add them to the library.

**Reply by CassyGu** [reply]
I see. But from report.XIC.tsv file, I do not know how to distinguish the matched and non-matched ones. So I can't write a script to automatically draw the full MS2 of individual precursor.

**Reply by vdemichev** [reply]
Everything there which has non-zero signal has been matched.
